tremor-script/src/pos.rs

```diff
@@ -46,6 +46,15 @@ impl std::ops::Sub for Location {
     }
 }
 
+impl std::ops::Add<char> for Location {
+    type Output = Location;
+    fn add(self, c: char) -> Self::Output {
+        let mut l = self; // copy
+        l.shift(c);
+        l
+    }
```

Comment:
We can avoid this copy since we take ownership

```suggestion
    fn add(mut self, c: char) -> Self::Output {
        self.shift(c);
        self
    }
```

Reply:
I am not sure this has the desired effect. I explicitly want a new instance and not mutate self. Honestly i am a bit confused by the copy semantics. Would a call like this: 

```rust
my_function(loc1, loc1 + 'a')
```

Make a copy of loc1 for the second argument?